realm/realm-library/src/main/java/io/realm/internal/Collection.java

```diff
@@ -348,6 +348,7 @@ public Collection(SharedRealm sharedRealm, LinkView linkView, SortDescriptor sor
         this.context = sharedRealm.context;
         this.table = linkView.getTable();
         this.context.addReference(this);
+        this.loaded = true;
```

Comment:
We should not initialize `loaded` field twice.
`private boolean loaded = false;` should be `private boolean loaded;` and assign in each constructor.